LocationUpdatesBackgroundKotlin/app/src/main/res/layout/fragment_location_update.xml

```diff
@@ -36,7 +36,7 @@
             android:layout_marginBottom="@dimen/margin_normal"
             android:scrollbars="vertical"
             android:textIsSelectable="true"
-            app:layout_constraintBottom_toTopOf="@id/startOrStopLocationUpdatesButton"
+            app:layout_constraintBottom_toTopOf="@+id/startOrStopLocationUpdatesButton"
```

Comment:
Why is this + necessary? This seems to be referencing the actual view in line 46.

Reply:
Right, but it is defined later, so I'm [forward referencing it](https://stackoverflow.com/a/31818690/1631197), i.e., if a widget you are referencing is defined later, you should use **`@+id`**.

I realized I accidentally deleted it in the last CL, and I didn't mean to. Android Studio creates this format when you use the layout editor with a ConstraintLayout.  

Some might argue it could be harder to debug, so I could go through and change all later references in the **`id`** and remove the **`+`**, e.g., `android:id="@id/startOrStopLocationUpdatesButton"`, but Studio sticks to leaving them in both, so I wanted to be consistent with that.

Reply:
Got it. I was not aware of that behavior of Studio.